### PR TITLE
PR for review + comments 4-12

### PR DESCRIPTION
**Pull request by MatthewDorner**

### CODE REVIEW

**Comment by MatthewDorner** [comment]
In general, there are 2 things that may cause problems between the old type of select helper and the new, ember-rapid-forms select helper. 1) new one automatically changes the given property on the model whenever user changes selection, while old one would only trigger an action when user changed selection. that's why many of the old ones say action=(action (mut property))  2) new one also seems to work with actions, but if provided an action, will also trigger the action on rendering the component for the first time, in addition to whenever user changes selection, which the old would not do. this has caused, for example, problems with the way the action in app/appointments/edit/controller.js works and is the reason for the changes to that file in my current code.
I think the most important thing now is to confirm that the above is correct (I'm basing it off my observations but I'm also pretty new to Ember.) and then decide what is the best way to deal with those differences.
You can see the code for the ember-rapid-forms select helper in https://github.com/piceaTech/ember-rapid-forms/blob/master/addon/components/em-select.js
Particularly at line 138 in _setValue, I see:
model.set(this.get('property'), selectedID);
     const changeAction = this.get('action');
     if(changeAction){
       changeAction(selectedID);
}
and that's in _setValue, which earlier in the code it looks like _setValue is explicitly being called when the component renders, and whenever the selection changes. so to me, this appears to support that it's behaving as I observed, and that it's supposed to behave that way.

**Comment by MatthewDorner** [comment]
also I will note that the majority of instances of select controls in the project are already using the ember-rapid-forms helper. the ones we're trying to change are just in a minority of those that for whatever reason are using the other one.

**Comment by MatthewDorner** [comment]
[Notes as of 4-15](https://github.com/MatthewDorner/hospitalrun-frontend/files/1913505/notes4-15.txt)

I have attached my notes, listing each instance of {{select-list}} to be replaced, and some of the things I observed while making the changes to specific ones. I'd greatly appreciate any input on this and the current code changes, in the meantime I am going to start looking at CSS.

File: app/components/quantity-conv.js [hunk 1]
```diff
@@ -12,6 +14,16 @@ export default Component.extend({
   targetUnit: alias('parentView.targetUnit'),
   unitList: null,
 
+  unitListHash: computed('unitList', function() {
+    let listArray = this.get('unitList');
+      return listArray.map((value) => {
```
Comment (efx) [inline app/components/quantity-conv.js]: NOTE: this is a syntactic sugar ES6 tidbit. When using fat arrow functions that have a single argument, you can omit the parentheses like so `listArray.map(val => val)`.
Additionally when arrow functions have a single line they return that implicitly. Many JS folk do this for brevity. There are some additional rules so returning a single value can sometimes require wrapping it in parentheses. So all that to say the above could be written as:
```js
listArray.map(val => ({ id: value, value }));
```
Clearly this is a matter of style and idiom.

File: app/templates/components/quantity-conv.hbs [hunk 2]
```diff
@@ -17,19 +17,19 @@
...
-        value=unit
-        action=(action (mut unit))
-        prompt=" "
-      }}
+      {{#em-form model=this submitButton=false as |form|}}
+        {{form.select
+          class='form-input-group test-inv-unit'
+          content=unitListHash
+          property='unit'
+          prompt=" "
```
Comment (efx) [inline app/templates/components/quantity-conv.hbs]: nitpick: these quotes should be single to match the other lines

**Comment by efx** [comment]
@MatthewDorner I still haven't built the app to manually go over the data binding issue you mentioned. Hopefully I can sometime this week.

**Comment by MatthewDorner** [comment]
Reviewed with @efx, did final checks and reverted debug code.

**Pull request merged**